### PR TITLE
Tags: Fix : remove default Input focus style

### PR DESCRIPTION
**Pull request by hepiyellow**
Fixes <Tags> bad styling caused by new Input focus style #1496

### CODE REVIEW

File: src/Input/Input.js [hunk 1]
```diff
@@ -169,9 +169,9 @@ class Input extends Component {
     this.input && this.input.select();
   };
 
-  _onFocus = () => {
+  _onFocus = e => {
```
Comment (hepiyellow) [inline src/Input/Input.js]: This was just missing. It is not necessary for this fix.

File: src/InputWithOptions/InputWithOptions.js [hunk 2]
```diff
@@ -197,15 +197,15 @@ class InputWithOptions extends WixComponent {
     }
   }
 
-  _onFocus() {
+  _onFocus(e) {
```
Comment (hepiyellow) [inline src/InputWithOptions/InputWithOptions.js]: This was just missing. It is not necessary for this fix.

File: src/MultiSelect/InputWithTags.js [hunk 3]
```diff
@@ -21,13 +21,19 @@ class InputWithTags extends React.Component {
     this.props.autoFocus && this.props.onFocus();
   }
 
-  handleInputFocus() {
+  handleClick(e) {
```
Comment (hepiyellow) [inline src/MultiSelect/InputWithTags.js]: This was previously called `handleInputFocus()`, but it was attached to the `onClick` event.

File: src/MultiSelect/InputWithTags.js [hunk 4]
```diff
@@ -21,13 +21,19 @@ class InputWithTags extends React.Component {
...
     this.input.focus();
-    this.setState({inputHasFocus: true});
+    this.props.onInputClicked && this.props.onInputClicked(e);
   }
 
-  handleInputBlur() {
-    this.setState({inputHasFocus: false});
+  handleInputFocus(e) {
+    !this.state.inputHasFocus && this.setState({inputHasFocus: true});
+    this.props.onFocus && this.props.onFocus(e);
```
Comment (hepiyellow) [inline src/MultiSelect/InputWithTags.js]: Added call to `props.onFocus`  which was missing.

File: src/MultiSelect/InputWithTags.js [hunk 5]
```diff
@@ -21,13 +21,19 @@ class InputWithTags extends React.Component {
     this.props.autoFocus && this.props.onFocus();
   }
 
-  handleInputFocus() {
+  handleClick(e) {
     this.input.focus();
-    this.setState({inputHasFocus: true});
+    this.props.onInputClicked && this.props.onInputClicked(e);
```
Comment (hepiyellow) [inline src/MultiSelect/InputWithTags.js]: Added call to `this.props.onInputClicked` which was missing. 
This fixes a feature of `InputWithOptions` (`MultiSelect`) which hide the options.

File: src/MultiSelect/InputWithTags.js [hunk 6]
```diff
@@ -21,13 +21,19 @@ class InputWithTags extends React.Component {
...
   }
 
-  handleInputBlur() {
-    this.setState({inputHasFocus: false});
+  handleInputFocus(e) {
+    !this.state.inputHasFocus && this.setState({inputHasFocus: true});
+    this.props.onFocus && this.props.onFocus(e);
+  }
+
+  handleInputBlur(e) {
```
Comment (hepiyellow) [inline src/MultiSelect/InputWithTags.js]: Previously this method only did:
`this.setState({inputHasFocus: false});`

File: src/MultiSelect/InputWithTags.js [hunk 7]
```diff
@@ -42,7 +48,19 @@ class InputWithTags extends React.Component {
...
+      'onManuallyInput',
+      'inputElement',
+      'closeOnSelect',
+      'predicate',
+      'menuArrow',
+      'onClickOutside',
+      'fixedHeader',
+      'fixedFooter',
+      'dataHook',
+      'onFocus',
```
Comment (hepiyellow) [inline src/MultiSelect/InputWithTags.js]: omit `onFocus` `onBlur` `onInputClicked`, so they don't get attached to the <Input>.
We attach own listeners instead, and delegate to the `props`'s listener.

File: src/MultiSelect/InputWithTags.js [hunk 8]
```diff
@@ -21,13 +21,19 @@ class InputWithTags extends React.Component {
...
-  handleInputFocus() {
+  handleClick(e) {
     this.input.focus();
-    this.setState({inputHasFocus: true});
+    this.props.onInputClicked && this.props.onInputClicked(e);
   }
 
-  handleInputBlur() {
-    this.setState({inputHasFocus: false});
+  handleInputFocus(e) {
```
Comment (argshook) [inline src/MultiSelect/InputWithTags.js]:
```
handleInputFocus = event => 
  this.setState(
    {inputHasFocus: true},
    () => this.props.onFocus(event)
  );

handleInputBlur = event => 
  this.setState(
    {inputHasFocus: false},
    () => this.props.onBlur(event)
  );
```

not to forget

```
defaultProps = {
  onBlur: i => i,
  onFocus: i => i
}
```

**Comment by hepiyellow** [comment]
Replaced by #1513

**Pull request closed without merging**